### DISCUSSION TITLE
Welcome to jitterphysics2 Discussions!

### DISCUSSION THREAD

**Discussion by notgiven688** [notgiven688/jitterphysics2#15] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together.

**Comment by Roderik11** [comment]
I've been using jitter for many private projects over the years and i'm happy to see a new version being worked on!
Two questions/suggestions:

1. Branch with double precision.
This would be nice to have as game worlds become larger and floating origin always feels iffy to work around.
I've been using a modified version of jitter (ported to double precision) for a real-scale space game and it helped me tremendously.

2. System.Numerics in favour of JVector
It's clear you've been aiming for max performance with your linear math, but System.Numerics has matured and offers performance as well as SIMD support. Perhaps consider using it over JVector.

**Reply by notgiven688** [reply]
Thank you for your feedback!

1. Your suggestion about double precision is compelling. In Jitter1, there were instances where the code failed in certain scenarios due to limited precision, with raycasting being a notable example. The current code handles limited precision much better. However, a double-precision version of the engine would indeed be a valuable addition. I will take this under advisement.

2. As for incorporating vector intrinsics for improved speed, this will certainly make its way into Jitter, albeit not immediately. And the introduction won't be as straightforward as merely replacing JVector and JMatrix with their counterparts from System.Numerics. Thus, JMatrix and JVector will likely remain.
